huaweicloud/services/vpc/resource_huaweicloud_vpc.go

```diff
@@ -201,26 +202,45 @@ func resourceVirtualPrivateCloudRead(_ context.Context, d *schema.ResourceData,
 	d.Set("routes", routes)
 
 	// save VirtualPrivateCloudV2 tags
-	if vpcV2Client, err := config.NetworkingV2Client(config.GetRegion(d)); err == nil {
-		if resourceTags, err := tags.Get(vpcV2Client, "vpcs", d.Id()).Extract(); err == nil {
-			tagmap := utils.TagsToMap(resourceTags.Tags)
-			if err := d.Set("tags", tagmap); err != nil {
-				return diag.Errorf("error saving tags to state for VPC (%s): %s", d.Id(), err)
-			}
-		} else {
-			log.Printf("[WARN] Error fetching tags of VPC (%s): %s", d.Id(), err)
-		}
-	} else {
-		return diag.Errorf("error creating VPC client: %s", err)
+	v2Client, err := conf.NetworkingV2Client(region)
+	if err != nil {
+		return diag.Errorf("error creating VPC v2 client: %s", err)
+	}
+
+	resourceTags, err := tags.Get(v2Client, "vpcs", d.Id()).Extract()
+	if err != nil {
+		return diag.Errorf("error fetching tags of VPC (%s): %s", d.Id(), err)
+	}
+
+	tagmap := utils.TagsToMap(resourceTags.Tags)
+	setTagErr := d.Set("tags", tagmap)
+	if setTagErr != nil {
+		return diag.Errorf("error saving tags to state for VPC (%s): %s", d.Id(), err)
+	}
+
+	// save VirtualPrivateCloudV3 extend_cidr
+	v3Client, err := conf.HcVpcV3Client(region)
+	if err != nil {
+		return diag.Errorf("error creating VPC v3 client: %s", err)
 	}
 
+	res, err := obtainV3VpcResp(v3Client, d.Id())
+	if err != nil {
+		diag.Errorf("error retrieving VPC (%s) v3 detail: %s", d.Id(), err)
+	}
+	var extendCidr string
+	for _, value := range res.Vpc.ExtendCidrs {
+		extendCidr = value
```

Comment:
it seems that we use the last item as secondary_cidr, why not using the first one?